src/pytorch_lightning/loggers/neptune.py

```diff
@@ -290,26 +288,27 @@ def __init__(
         self._api_key = api_key
         self._run_instance = run
         self._neptune_run_kwargs = neptune_run_kwargs
-        self._run_short_id = None
+        self._run_short_id: Optional[str] = None
 
         if self._run_instance is not None:
             self._retrieve_run_data()
 
             # make sure that we've log integration version for outside `Run` instances
-            self._run_instance[_INTEGRATION_VERSION_KEY] = __version__
+            self._run_instance[_INTEGRATION_VERSION_KEY] = pl.__version__
 
-    def _retrieve_run_data(self):
+    def _retrieve_run_data(self) -> None:
         try:
-            self._run_instance.wait()
-            self._run_short_id = self._run_instance["sys/id"].fetch()
-            self._run_name = self._run_instance["sys/name"].fetch()
+            if self._run_instance is not None:
+                self._run_instance.wait()
+                self._run_short_id = self._run_instance["sys/id"].fetch()
+                self._run_name = self._run_instance["sys/name"].fetch()
```

Comment:
```suggestion
            assert self._run_instance is not None
            self._run_instance.wait()
            self._run_short_id = self._run_instance["sys/id"].fetch()
            self._run_name = self._run_instance["sys/name"].fetch()
```
I'd not change the logic.